### DISCUSSION TITLE
Console Warnings and primitive using error

### DISCUSSION THREAD

**Discussion by khaledOghli** [Tresjs/tres#371] [Q&A] [answered]
Thank you so much for supporting three Js in Vue. I'm trying to use it with Nuxt.
but I have some issues.
When I used GLTFModel, everything worked well.
but I get these warnings.
How do I get rid of them?

1 - Failed to resolve component: primitive
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <Component path="/assets/model/rhetorician/scene.gltf" draco="" position= Array(3)  ... > 
  at <T>

2- THREE.WebGLRenderer: The property .useLegacyLights has been deprecated. Migrate your lighting according to the following guide: https://discourse.threejs.org/t/updates-to-lighting-in-three-js-r155/53733.

<img width="1582" alt="Screenshot 2023-08-18 at 2 54 17 PM" src="https://github.com/Tresjs/tres/assets/65956317/c3b888fa-4043-492e-ad41-878f41bb17e0">

<img width="1554" alt="Screenshot 2023-08-18 at 2 56 59 PM" src="https://github.com/Tresjs/tres/assets/65956317/7c57b4a2-5677-4f5d-befc-f39d0e0609e2">

The second issue is when I try to use primitive :object="nodes.MyModel"  instead of GLTFModel.
It does not work.
I get these errors.

<img width="1565" alt="Screenshot 2023-08-18 at 2 57 14 PM" src="https://github.com/Tresjs/tres/assets/65956317/23a91251-0f73-417e-b547-735df14b3d6f">

my code : 

`
<script setup lang="ts">
import { useGLTF } from '@tresjs/cientos'

const { scene, nodes, animations, materials } = await useGLTF('/assets/model/rhetorician/scene.gltf', { draco: true })

gsap.registerPlugin(ScrollTrigger)
const rotateModalX = ref(0)
const rotateModalY = ref(0)
const rotateModalZ = ref(0)
const lightRef = ref(null)

</script>

<template>
  <TresCanvas
    shadows alpha
    :controls="{ enableDamping: true, dampingFactor: 0.05 }"
  >
    <TresPerspectiveCamera
      :position="[0, 1, 4.5]"
      :fov="75"
      :near="0.5"
      :far="1000"
    />
    <OrbitControls />
    <Suspense>
      <GLTFModel
        path="/assets/model/rhetorician/scene.gltf" draco
        :position="[0, -1.5, 0]"
        :scale="[1.1, 1, 1]"
        :rotation="[rotateModalX, rotateModalY, rotateModalZ]"
        :cast-shadow="true"
      />
      
    </Suspense>
    
    

  </TresCanvas>
</template>

`

**Comment by alvarosabu** [comment]
Hey @khaledOghli thanks for reaching out

Are you using the nuxt module? https://github.com/Tresjs/nuxt

**Reply by khaledOghli** [reply]
Yes I'm using the nuxt module

**Comment by alvarosabu** [comment, marked as answer]
> THREE.WebGLRenderer: The property .useLegacyLights has been deprecated. Migrate your lighting according to the following guide: https://discourse.threejs.org/t/updates-to-lighting-in-three-js-r155/53733.

This is because of the threejs v155 update, you can ignore it

**Comment by khaledOghli** [comment]
Awesome Man I Love Tres js @alvarosabu , and everything works well, Just I go deep into the documentation,
on how to use TresDirectionalLightHelper and I'm not able to find about light in docs.
any suggestion or advice !
